nav2_amcl/src/sensors/laser/beam_model.cpp

```diff
@@ -72,6 +72,12 @@ BeamModel::sensorFunction(LaserData * data, pf_sample_set_t * set)
     step = (data->range_count - 1) / (self->max_beams_ - 1);
     for (i = 0; i < data->range_count; i += step) {
       obs_range = data->ranges[i][0];
+
+      // Check for NaN
+      if (obs_range != obs_range) {
```

Comment:
Actually check for NaNs properly, this is a really not-awesome way of doing it. https://cplusplus.com/reference/cmath/isnan/

Reply:
Sure thing. See https://github.com/ros-planning/navigation2/pull/3929/commits/8ac39c06d0285f46389828b248806a26528737de.